nemo/collections/asr/parts/perturb.py

```diff
@@ -486,3 +756,67 @@ class CustomPerturbation(perturb.Perturbation):
 
     augmenter = AudioAugmentor(perturbations=augmentations)
     return augmenter
+
+
+class AugmentationDataset(IterableDataset):
+    """
+        A class that loads tarred audio files and cycles over the files in the dataset.
+
+        Accepts a single comma-separated JSON manifest file (in the same style as for the AudioToCharDataset/AudioToBPEDataset),
+        as well as the path(s) to the tarball(s) containing the wav files. Each line of the manifest should
+        contain the information for one audio file, including at least the transcript and name of the audio
+        file within the tarball.
+
+        Valid formats for the audio_tar_filepaths argument include:
+        (1) a single string that can be brace-expanded, e.g. 'path/to/audio.tar' or 'path/to/audio_{1..100}.tar.gz', or
+        (2) a list of file paths that will not be brace-expanded, e.g. ['audio_1.tar', 'audio_2.tar', ...].
+
+        Note: For brace expansion in (1), there may be cases where `{x..y}` syntax cannot be used due to shell interference.
+        This occurs most commonly inside SLURM scripts. Therefore we provide a few equivalent replacements.
+        Supported opening braces - { <=> (, [, < and the special tag _OP_.
+        Supported closing braces - } <=> ), ], > and the special tag _CL_.
+        For SLURM based tasks, we suggest the use of the special tags for ease of use.
+
+        See the WebDataset documentation for more information about accepted data and input formats.
+    """
+
+    def __init__(self, manifest_path: str, tar_filepaths: Union[str, List[str]], shuffle_n: int = 128):
+        self._manifest = collections.ASRAudioText(manifest_path, parser=parsers.make_parser([]), index_by_file_id=True)
+
+        if isinstance(tar_filepaths, str):
+            # Replace '(' and '[' with '{'
+            brace_keys_open = ['(', '[', '<', '_OP_']
+            for bkey in brace_keys_open:
+                if bkey in tar_filepaths:
+                    tar_filepaths = tar_filepaths.replace(bkey, "{")
+
+            # Replace ')' and ']' with '}'
+            brace_keys_close = [')', ']', '>', '_CL_']
+            for bkey in brace_keys_close:
+                if bkey in tar_filepaths:
+                    tar_filepaths = tar_filepaths.replace(bkey, "}")
```

Comment:
You need a brace expander here after this line to properly expand the {} syntax.

Reply:
@titu1994 webdataset takes care of expanding the braces https://github.com/tmbdev/webdataset/blob/244357e68b538f38029ef3b0a5d1464e45aade41/webdataset/dataset.py#L438. I have tested this code and it works. 
We had to do it in TarredAudioToTextDataset because we wanted to assign only a certain number of shards to each gpu. Here we don't really care about it we just want the next noise sample from the dataset and all processes see all the shards.

Reply:
Ah sounds good then.